reference-implementation/src/header.rs

```diff
@@ -0,0 +1,149 @@
+/// An SFrame key ID
+#[derive(PartialEq, Eq, Hash, Debug, Copy, Clone)]
+pub struct KeyId(pub u64);
+
+/// An SFrame counter
+#[derive(PartialEq, Eq, Hash, Debug, Copy, Clone)]
+pub struct Counter(pub u64);
+
+/// The SFrame header.  It stores KID and CTR values for a ciphertext, and the encoded form.
+#[derive(PartialEq, Eq, Debug)]
+pub struct Header {
+    /// The key ID for the key with which the ciphertext was encrypted
+    pub kid: KeyId,
+
+    /// The counter used to form a unique nonce for the ciphertext
+    pub ctr: Counter,
+
+    encoded: Vec<u8>,
+}
+
+fn to_min_be_bytes(val: u64) -> Vec<u8> {
+    let bytes = val.to_be_bytes();
+    let start = bytes.iter().position(|&x| x != 0).unwrap_or(7);
```

Comment:
You can use leading_zeros() for calculating this: `min((u64::BITS - val.leading_zeros()) / 8, 7)`